c/doc/run_arduino.md

```diff
@@ -0,0 +1,91 @@
+# Arduino Yun Setup
+
+The following document describes the process of connecting an Arduino Yun system to Azure IoT Hub.
+
+## Table of Contents
+
+- [Requirements](#requirements)
+- [Before you begin](#beforebegin)
+- [Setup the Arduino Yun Board](#setupboard)
+- [Setup the development environment](#setup)
+- [Build and run the sample](#buildrunapp)
+
+<a name="requirements"/>
+## Requirements
+
+  - Computer with a Git client installed so that you can access the azure-iot-sdks code on GitHub.
+
+<a name="beforebegin"/>
+## Before you begin
+
+Before you begin you will need to create and configure an IoT hub to connect to.
+  1. [Set up your IoT Hub][setup-iothub].
+  1. With your IoT hub configured and running in Azure, follow the instructions in [Connecting your device to an IoT hub][provision-device].
+  1. Make note of the Connection String for your device from the previous step.
+
+  > Note: You can skip this step if you just want to build the sample application without running it.
+
+<a name="setupboard"/>
+## Setup the Arduino Yun board
+This document assumes that your development board is upgraded to Linino. If you have done so, please refer to this document
+
+http://labs.arduino.org/How+to+upgrade+an+old+Arduino+Yun+to+LininoIO
+
+<a name="setup"/>
+## Setup the development environment
+
+This section shows you how to set up a development environment for the Azure IoT device SDK for C on Arduino Yun.
+
+1. Clone this repository ([azure-iot-sdks](https://github.com/Azure/azure-iot-sdks)) to the machine you are using.
+2. Open a shell and navigate to the folder **c/build_all/arduino** in your local copy of the repository.
+3. If you have not installed the OpenWRT SDK, run `setup_sdk.sh`.
+3. Run the `setup.sh` script to install the prerequisite packages and the dependent libraries.
+4. Add to the top of the file `src/proton-c/CMakeLists.txt` the following contents:
+
+SET(IN_OPENWRT $ENV{IN_OPENWRT})
```

Comment:
This would require modifying the proton repo as we are pulling from https://github.com/apache/qpid-proton/. I am assuming that this is not ideal, correct?